pkg/csi_driver/controller.go

```diff
@@ -975,9 +984,13 @@ func (s *controllerServer) CreateSnapshot(ctx context.Context, req *csi.CreateSn
 			},
 		}
 		klog.V(4).Infof("CreateSnapshot succeeded for volume %v, Backup Id: %v", volumeID, backupObj.Name)
-		return resp, nil
 	}
 
+	if err := s.config.tagManager.AttachResourceTags(ctx, cloud.FilestoreBackUp, backupInfo.Name, backupInfo.Location, req.GetName(), req.GetParameters()); err != nil {
+		return nil, file.StatusError(err)
```

Comment:
If tagManager returns 429, this will convert it to ResourceExhausted. ResourceExhausted is only retried in deferred provisioning scenario: https://github.com/kubernetes-csi/external-provisioner/blob/d981ff3867118eeb37be7b75ba4594a7aa18e301/pkg/controller/controller.go#L846

I think you want to return Unavailable in all scenarios, to allow retry.

Reply:
@pwschuurman ah understood. Thank you for the reference!
suggestion incorporated.